src/bun.js/webcore/body.zig

```diff
@@ -1258,3 +1263,322 @@ pub fn BodyMixin(comptime Type: type) type {
         }
     };
 }
+
+pub const BodyValueBufferer = struct {
+    const log = bun.Output.scoped(.BodyValueBufferer, false);
+
+    const ArrayBufferSink = JSC.WebCore.ArrayBufferSink;
+    const Callback = *const fn (ctx: *anyopaque, bytes: []const u8, err: ?JSC.JSValue, is_async: bool) void;
+
+    ctx: *anyopaque,
+    onFinishedBuffering: Callback,
+
+    js_sink: ?*ArrayBufferSink.JSSink = null,
+    byte_stream: ?*JSC.WebCore.ByteStream = null,
```

Comment:
it's not always a ByteStream? Is this specific to fetch() and only fetch()?

Reply:
js_sink for `.Direct` and `.Javascript` + byte_stream for `.Bytes`, this was inspired by `Bun.serve` code, should not be specific to only `fetch`, `.Blob, .File` should be handled by `toBlobIfPossible` if the comment is true